src/passport-saml/multiSamlStrategy.ts

```diff
@@ -8,35 +8,35 @@ import {
   AuthorizeOptions,
   MultiSamlConfig,
   RequestWithUser,
+  SamlConfig,
   VerifyWithoutRequest,
   VerifyWithRequest,
 } from "./types";
 
 class MultiSamlStrategy extends SamlStrategy {
-  _options: MultiSamlConfig;
+  static readonly newSamlProviderOnConstruct = false;
+
+  _options: SamlConfig & MultiSamlConfig;
 
   constructor(options: MultiSamlConfig, verify: VerifyWithRequest);
   constructor(options: MultiSamlConfig, verify: VerifyWithoutRequest);
   constructor(options: MultiSamlConfig, verify: never) {
-    if (!options || typeof options.getSamlOptions != "function") {
+    if (!options || typeof options.getSamlOptions !== "function") {
       throw new Error("Please provide a getSamlOptions function");
     }
 
-    if (!options.requestIdExpirationPeriodMs) {
-      options.requestIdExpirationPeriodMs = 28800000; // 8 hours
-    }
-
-    if (!options.cacheProvider) {
```

Comment:
It seems like this breaks the default `InMemoryCacheProvider` for `MultiSamlStrategy` added via #350  (discussed in #334)